### ISSUE TITLE
Circular drive messes up if the root parent object is rotated.

### ISSUE THREAD

**Issue by DeltaMike1010-driya** [ValveSoftware/steamvr_unity_plugin#461]
Hello everyone, I am making a **Forklift simulation** where you use controllers to drive it around. The user has to grab and drag **(grab-grip) a lever to toggle between Acceleration and Reverse modes**, and a steering wheel which is grabbed and rotated to steer using grab pinch on left hand. The lever works fine if the **vehicle's rotation is similar to the start rotation** of the vehicle. After making a turn or few, the **axis changes from X to Z**. So I have to grab and drag the lever in **Z-axis to rotate it on X-axis** (the lever is supposed to be grabbed and rotated on the X-axis). I'm assuming that this is because of the `worldPlaneNormal `being calculated only in `Start()`. I tried updating the `worldPlaneNormal `every time `ComputeAngle() `is called, but resulted in twitchy rotation. Surprisingly, I also have a steering wheel which uses the `CircularDrive `component but it works like a charm. Steering wheel uses the Y-axis for rotation.

Is there a work-around for this?

**Comment by DeltaMike1010-driya** [comment]
I also tried removing the player prefab from the vehicle hierarchy and made it follow the vehicle instead, but this issue still persists. Any other **rotation** of vehicle ** (essentially the root object under which the circular drive is attached to steering and lever) other than start rotation** results in **unnatural behavior of Circular drive**. I also observed that the steering too is not working properly. 

Is there anything I can do to fix this?
